### ISSUE TITLE
R2

### ISSUE THREAD

**Issue by cblgg** [Mohamedelrefaie/DrivAerNet#13]
![图片1](https://github.com/user-attachments/assets/fa85f9c9-2095-4089-948a-c5ac6d09d587)

And “ Epoch 78Training Loss: 0.079081 Time: 4061.08s
Average Training R²: -368.4631”
“Epoch 78 Validation Loss: 0.0161, Avg Inference Time: 0.3453s
Average Validation R²: -58.6274”parameters are: 'exp_name': 'CdPrediction_DrivAerNet_r2_100epochs_5k',
'cuda': True,
'seed': 1,
'num_points': 5000,
'lr': 0.001,
'batch_size': 4,
'epochs': 100,
'dropout': 0.4,
'emb_dims': 512,
'k': 40,
'optimizer': 'adam',

**Comment by Mohamedelrefaie** [comment]
Thank you for reaching out and providing the training details. I suggest increasing the batch size, which might make it easier to debug and help optimize memory usage. Additionally, I recommend caching the point clouds offline to speed up training significantly. Seems like your training is very slow 

We reran the experiments and obtained the same results as reported in the NeurIPS paper. However, after some simple optimizations, we were able to achieve even better results than initially reported.

**Comment by phanindrasp** [comment]
I have the exact same issue even with a batch size of 8. Let me know if you've any updates @cblgg

**Comment by brooks-Miles** [comment]
> Thank you for reaching out and providing the training details. I suggest increasing the batch size, which might make it easier to debug and help optimize memory usage. Additionally, I recommend caching the point clouds offline to speed up training significantly. Seems like your training is very slow
> 
> We reran the experiments and obtained the same results as reported in the NeurIPS paper. However, after some simple optimizations, we were able to achieve even better results than initially reported.

Hello, author. Here are the results I obtained using the following parameters:
'cuda': True,
'seed': 1,
'num_points': 5000,
'lr': 0.001,
'batch_size': 32,
'epochs': 100,
'dropout': 0.5,
'emb_dims': 512,
'k': 40.
The results obtained:
Testing the final model:
Test MSE: 0.000994，Test MAE: 0.027736，Max MAE: 0.032676，Test R²: -0.4203
Total inference time: 0.34s for 576 samples
Testing the best model:
Test MSE: 0.000172，Test MAE: 0.010169，Max MAE: 0.013186，Test R²: 0.7665
Total inference time: 0.58s for 576 samples
Here is the loss curve：
![损失曲线](https://github.com/user-attachments/assets/d77650cd-05e3-4eec-85ed-d7767fcbb5ab)
Even though I increased the number of epochs, the R² value of the model does not reach 0.901 after convergence. I would appreciate your guidance. Thank you very much.

**Comment by brooks-Miles** [comment]
> > Thank you for reaching out and providing the training details. I suggest increasing the batch size, which might make it easier to debug and help optimize memory usage. Additionally, I recommend caching the point clouds offline to speed up training significantly. Seems like your training is very slow
> > We reran the experiments and obtained the same results as reported in the NeurIPS paper. However, after some simple optimizations, we were able to achieve even better results than initially reported.
> 
> Hello, author. Here are the results I obtained using the following parameters: 'cuda': True, 'seed': 1, 'num_points': 5000, 'lr': 0.001, 'batch_size': 32, 'epochs': 100, 'dropout': 0.5, 'emb_dims': 512, 'k': 40. The results obtained: Testing the final model: Test MSE: 0.000994，Test MAE: 0.027736，Max MAE: 0.032676，Test R²: -0.4203 Total inference time: 0.34s for 576 samples Testing the best model: Test MSE: 0.000172，Test MAE: 0.010169，Max MAE: 0.013186，Test R²: 0.7665 Total inference time: 0.58s for 576 samples Here is the loss curve： ![损失曲线](https://private-user-images.githubusercontent.com/188964597/396495394-d77650cd-05e3-4eec-85ed-d7767fcbb5ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0Mzg4NjIsIm5iZiI6MTczNDQzODU2MiwicGF0aCI6Ii8xODg5NjQ1OTcvMzk2NDk1Mzk0LWQ3NzY1MGNkLTA1ZTMtNGVlYy04NWVkLWQ3NzY3ZmNiYjVhYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxN1QxMjI5MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iY2Q5ZDczNTFjZGQ5YjI0ZjVjMWNmZmVlNjE2MWFkYWU5YjYzNzk0ZWYyOTJjOWMyMzdiMmNhMGNkMTY1ZGVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.U3fI460FwryWVQBPInoJRGut4xC7EFfUrcO0fsoHkNs) Even though I increased the number of epochs, the R² value of the model does not reach 0.901 after convergence. I would appreciate your guidance. Thank you very much.



> Thank you for reaching out and providing the training details. I suggest increasing the batch size, which might make it easier to debug and help optimize memory usage. Additionally, I recommend caching the point clouds offline to speed up training significantly. Seems like your training is very slow
> 
> We reran the experiments and obtained the same results as reported in the NeurIPS paper. However, after some simple optimizations, we were able to achieve even better results than initially reported.

The model I trained is RedgCNN, and the dataset used is DrivAerNet_STLs_Combined.I am using the raw STL files.
